lib/src/data/communication/api/api_errors.dart

```diff
@@ -0,0 +1,33 @@
+abstract class ApiError {
```

Comment:
jeżeli jest to data, to nie powinno przenikać do ani do warstw domain ani do ui, także imo to powinna być warstwa domeny